src/lib/ViZDoomController.cpp

```diff
@@ -291,12 +295,29 @@ namespace vizdoom {
         }
 
         int ticsMade = 0;
-
+        this->largerAudioBuffer = NULL;
         for (unsigned int i = 0; i < tics; ++i) {
-            if (i == tics - 1) this->tic(update);
+            if (i == tics - 1) this->tic(true);
             else this->tic(false);
 
-            ++ticsMade;
+            if (!this->noSound) {
+                size_t audioSize = static_cast<size_t>(audioLength * 2);
+                short *abuf = this->getAudioBuffer();
+
+                const std::shared_ptr<std::vector<short>> &temp_aud = std::make_shared<std::vector<short>>(abuf, abuf +
+                                                                                                                 audioSize);
+                if ((this->largerAudioBuffer == NULL)) {
+                    this->largerAudioBuffer = temp_aud;
+                } else {
+                    for (unsigned int j = 0; j < (audioLength * 2); j++) {
+                        this->largerAudioBuffer->push_back(temp_aud->data()[j]);
```

Comment:
Also, this is where we can easily change the logic to decouple frameskip from the number of audio frames.
Instead of resetting largeAudioBuffer to NULL at the beginning of every tick, we can use `deque` http://www.cplusplus.com/reference/deque/deque/

If we pre-allocate it to the max size we want to hold and populate it with zeros initially.

Then on every tick we `emplace_back` the new audio ticks, and then `pop_front` to remove extra elements from the front and keep the deque at desired size. 

This will allow us to always have up to MAX_AUDIO_LENGTH frames in the deque, regardless how many frames we actually advanced.

Reply:
yup, fixed. Still working on the decoupling part

Reply:
Should be decoupled now